spec/avram/operations/save_operation_spec.cr

```diff
@@ -307,6 +307,24 @@ describe "Avram::SaveOperation" do
     end
   end
 
+  describe ".bulk_upsert" do
+    context "when the records are not persisted" do
+      it "should create the records" do
+        record_args = (1..2).to_a.map do |i|
+          {
+            name:     "Test #{i}",
+            nickname: "Test Nickname #{i}",
+          }
+        end
+
+        UpsertUserOperation.bulk_upsert(record_args)
```

Comment:
Since this is just taking an Array of NamedTuple, there doesn't seem to be any safety here. For example, what if I add a key for a column that doesn't exist?

Reply:
oh sweet! Ok